donkeycar/contrib/gradcam/cli.py

```diff
@@ -0,0 +1,432 @@
+import cv2
+import tensorflow as tf
+from tensorflow.keras.models import load_model
+
+from .utils import (
+    find_last_conv_layer,
+    make_gradcam_heatmap,
+    create_gradcam_visualization
+)
+
+
+def parse_args():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(
+        description='Generate Grad-CAM visualizations for DonkeyCar models'
+    )
+
+    # Required arguments
+    parser.add_argument(
+        '--model_path',
+        required=True,
+        help='Path to the trained model (.h5 file)'
+    )
+
+    parser.add_argument(
+        '--data_path',
+        required=True,
+        help='Path to the dataset directory containing images'
+    )
+
+    # Optional arguments
+    parser.add_argument(
+        '--output_dir',
+        default='gradcam_output',
+        help='Directory to save output visualizations'
+    )
+
+    parser.add_argument(
+        '--video_name',
+        default='grad_cam_visualization',
+        help='Name for the output video file'
+    )
+
+    parser.add_argument(
+        '--fps',
+        type=int,
+        default=20,
+        help='Frames per second for the output video'
+    )
+
+    parser.add_argument(
+        '--stride',
+        type=int,
+        default=1,
+        help='Process every Nth frame'
+    )
+
+    parser.add_argument(
+        '--max_frames',
+        type=int,
+        default=None,
+        help='Maximum number of frames to process'
+    )
+
+    parser.add_argument(
+        '--alpha',
+        type=float,
+        default=0.4,
+        help='Transparency of the heatmap overlay (0.0-1.0)'
+    )
+
+    parser.add_argument(
+        '--exclude_regions',
+        type=str,
+        default=None,
+        help='Regions to exclude from heatmap (x1,y1,x2,y2|x1,y1,x2,y2|...)'
+    )
+
+    return parser.parse_args()
+
+
+def load_model_safely(model_path):
+    """
+    Load a Keras model with appropriate error handling.
+
+    Args:
+        model_path: Path to the model file
+
+    Returns:
+        Loaded Keras model
+    """
+    print(f"Loading model from {model_path}")
+
+    try:
+        # Try to load the model with custom objects to handle loss functions
+        custom_objects = {
+            'mse': tf.keras.losses.mean_squared_error,
+            'mean_squared_error': tf.keras.losses.mean_squared_error
+        }
+        model = load_model(model_path, custom_objects=custom_objects)
+        print("Model loaded successfully with custom objects")
+        return model
+    except Exception as e:
+        print(f"Error loading model with custom objects: {e}")
+
+        try:
+            # Alternative loading method with compile=False
+            model = load_model(model_path, compile=False)
+            print("Model loaded successfully with compile=False")
+            return model
+        except Exception as e2:
+            print(f"Fatal error loading model: {e2}")
+            sys.exit(1)
+
+
+def find_images(data_path):
+    """
```

Comment:
This will not be needed when going through the `MakeMovie` class as all the data preparation is already in place.